labgrid/driver/shelldriver.py

```diff
@@ -137,37 +163,32 @@ def _await_login(self):
             elif index == 1:
                 # we need to login
                 self.console.sendline(self.username)
-                index, _, _, _ = self.console.expect([self.prompt, "Password: "], timeout=10)
-                if index == 1:
-                    if self.password:
-                        self.console.sendline(self.password)
-                        remaining_time = (start + self.login_timeout) - time.time()
-                        self.console.expect(self.prompt, timeout=remaining_time)
-                    else:
-                        raise Exception("Password entry needed but no password set")
+
+                # Do not issue the check string until the password has been entered
+                check = not self.password
+                for index in poll_expect([self.prompt, "Password: "], check):
```

Comment:
I'm not sure if we really need a nested poll just to handle a password. We could also roll the password detection as just another case in the main poll loop.

Anyone have thoughts?

Reply:
We do this inner loop to decide if we actually need to enter the password after the login prompt. IMO this could be folded into the main loop.

Reply:
This was resolved by @JoshuaWatt in a follow-up commit.